src/libsystemd/sd-event/sd-event.c

```diff
@@ -1037,32 +1037,33 @@ static int event_setup_timer_fd(
                 struct clock_data *d,
                 clockid_t clock) {
 
-        int r, fd;
-
         assert(e);
         assert(d);
 
         if (_likely_(d->fd >= 0))
                 return 0;
 
+        _cleanup_close_ int fd = -1;
+        int r;
+
         fd = timerfd_create(clock, TFD_NONBLOCK|TFD_CLOEXEC);
         if (fd < 0)
                 return -errno;
 
         fd = fd_move_above_stdio(fd);
+        if (fd < 0)
+                return fd;
```

Comment:
see the comment in fd_move_move_stdio()...

This function is actually written in a style so that it never fails. i.e. if the operation fails it just returns the original fd. it's a safety precaution after all only, to avoid stepping into the territory of fd0/1/2 needlessly.

Reply:
if you want we could rename the call to fd_move_above_stdio_never_fail() or so, maybe this would help to not run into this trap when reading this again years after?

Reply:
I looked at `fd_move_above_stdio` and saw that it allocates an fd, and assumed that it might fail. My bad. I should have read the comment ;) I think the name is OK.